app/services/stats/fee_scheme_usage_generator.rb

```diff
@@ -0,0 +1,154 @@
+require 'csv'
+
+module Stats
+  class FeeSchemeUsageGenerator
+    CLAIM_TYPES =
+      %w[Claim::AdvocateClaim
+         Claim::AdvocateHardshipClaim
+         Claim::AdvocateInterimClaim
+         Claim::AdvocateSupplementaryClaim
+         Claim::InterimClaim
+         Claim::LitigatorClaim
+         Claim::LitigatorHardshipClaim
+         Claim::TransferClaim].freeze
+
+    include StuffLogger
+
+    def self.call(...)
+      new(...).call
+    end
+
+    def initialize(**kwargs)
+      @format = kwargs.fetch(:format, :csv)
+    end
+
+    def call
+      log_info('Fee Scheme Usage Report generation started...')
+      output = generate_new_report
+      log_info('Fee Scheme Usage Report generation finished')
+      Stats::Result.new(output, @format)
+    end
+
+    private
+
+    def ordered_fee_schemes
+      @ordered_fee_schemes ||= FeeScheme.where(name: 'AGFS')
+                                        .map { |scheme| "#{scheme.name} #{scheme.version}" }
+                                        .sort_by { |x| x[/\d+/].to_i } +
+                               FeeScheme.where(name: 'LGFS')
+                                        .map { |scheme| "#{scheme.name} #{scheme.version}" }
+                                        .sort_by { |x| x[/\d+/].to_i }
```

Comment:
You can sort in the database query, which would also enable you to avoid two queries;

```suggestion
      @ordered_fee_schemes ||= FeeScheme.where(name: ['AGFS', 'LGFS']).order(:name, :version)
                                        .map { |scheme| "#{scheme.name} #{scheme.version}" }
```

Reply:
Thanks Joe! I didn't realise that you could put multiple conditions to sort by into the order method like that. I've made a new commit using this code (with the array tweaked to use %w[] to make rubocop happy)